test/DerivativeTest.jl

```diff
@@ -113,4 +114,16 @@ end
     @test ForwardDiff.derivative(x -> (1+im)*x, 0) == (1+im)
 end
 
+@testset "non-standard numbers" begin
+    Base.exp(f::Furlongs.Furlong) = exp(f.val)
+    Base.cos(f::Furlongs.Furlong) = cos(f.val)
+    Base.sin(f::Furlongs.Furlong) = sin(f.val)
```

Comment:
Similar to unitful quantities, shouldn't these only be defined for dimensionless furlongs?
```suggestion
    Base.exp(f::Furlongs.Furlong{0}) = exp(f.val)
    Base.cos(f::Furlongs.Furlong{0}) = cos(f.val)
    Base.sin(f::Furlongs.Furlong{0}) = sin(f.val)
```

Reply:
Undid this, at it made the tests fail